### ISSUE TITLE
Eq and Ord instances for Value

### ISSUE THREAD

**Issue by lehins** [lehins/massiv#19]
`Stencil` creation goes into great trouble in order to guarantee safety, so that created stencil can be applied to any array and there will be no silent memory reads that aren't allowed. This is the whole reason for existence of  `Value` data type in the first place, which prevents dependency of indices used by stencils on elements of the array it is being applied to, or so I thought. Apparently `Ord` and `Eq` instances for `Value` can cause havoc for stencil validation and should be removed. Consider example:

```
λ> let s = makeStencil (Fill 200) 3 1 $ \f -> if f 0 == f 1 then f (-1) else f 100
λ> let arr = mapStencil s $ computeAs P $ range Seq 0 10
λ> computeAs P arr
(Array P Seq (10)
  [ 200,140192713198744,8589934594,140192651898908,283614252664,1,140192717508584,1,200,200 ])
```

This is a breaking change and will be postponed to `v0.2`

**Comment by lehins** [comment]
@fosskers This issue will affect your `mapalgebra`, so it might be a good idea to switch to `forStencilUnsafe`, wherever you rely on `Eq`. I plan to have `v0.2` out sometime in the middle of the summer, so it won't affect you until then.

**Comment by fosskers** [comment]
I'm probably good, actually. I end up using the `Applicative` instance of `Value` a lot to get the fine-grained access that I need in certain places.

**Comment by lehins** [comment]
Fixed in 35fb9e5613ca604d9f7d3394f856cfab58fe0c0b
